subprojects/packagefiles/fdk-aac/meson.build

```diff
@@ -0,0 +1,403 @@
+  'libMpegTPEnc/src/tp_version.h',
+  'libMpegTPEnc/src/tpenc_adif.h',
+  'libMpegTPEnc/src/tpenc_adts.h',
+  'libMpegTPEnc/src/tpenc_asc.h',
+  'libMpegTPEnc/src/tpenc_latm.h',
+  'libSACdec/src/sac_bitdec.h',
+  'libSACdec/src/sac_calcM1andM2.h',
+  'libSACdec/src/sac_dec.h',
+  'libSACdec/src/sac_dec_conceal.h',
+  'libSACdec/src/sac_dec_interface.h',
+  'libSACdec/src/sac_dec_ssc_struct.h',
+  'libSACdec/src/sac_process.h',
+  'libSACdec/src/sac_qmf.h',
+  'libSACdec/src/sac_reshapeBBEnv.h',
+  'libSACdec/src/sac_rom.h',
+  'libSACdec/src/sac_smoothing.h',
+  'libSACdec/src/sac_stp.h',
+  'libSACdec/src/sac_tsd.h',
+  'libSACenc/src/sacenc_bitstream.h',
+  'libSACenc/src/sacenc_const.h',
+  'libSACenc/src/sacenc_delay.h',
+  'libSACenc/src/sacenc_dmx_tdom_enh.h',
+  'libSACenc/src/sacenc_filter.h',
+  'libSACenc/src/sacenc_framewindowing.h',
+  'libSACenc/src/sacenc_huff_tab.h',
+  'libSACenc/src/sacenc_nlc_enc.h',
+  'libSACenc/src/sacenc_onsetdetect.h',
+  'libSACenc/src/sacenc_paramextract.h',
+  'libSACenc/src/sacenc_staticgain.h',
+  'libSACenc/src/sacenc_tree.h',
+  'libSACenc/src/sacenc_vectorfunctions.h',
+  'libSBRdec/src/env_calc.h',
+  'libSBRdec/src/env_dec.h',
+  'libSBRdec/src/env_extr.h',
+  'libSBRdec/src/hbe.h',
+  'libSBRdec/src/HFgen_preFlat.h',
+  'libSBRdec/src/huff_dec.h',
+  'libSBRdec/src/lpp_tran.h',
+  'libSBRdec/src/psbitdec.h',
+  'libSBRdec/src/psdec.h',
+  'libSBRdec/src/psdec_drm.h',
+  'libSBRdec/src/pvc_dec.h',
+  'libSBRdec/src/sbr_deb.h',
+  'libSBRdec/src/sbr_dec.h',
+  'libSBRdec/src/sbr_ram.h',
+  'libSBRdec/src/sbr_rom.h',
+  'libSBRdec/src/sbrdec_drc.h',
+  'libSBRdec/src/sbrdec_freq_sca.h',
+  'libSBRdec/src/transcendent.h',
+  'libSBRenc/src/bit_sbr.h',
+  'libSBRenc/src/cmondata.h',
+  'libSBRenc/src/code_env.h',
+  'libSBRenc/src/env_bit.h',
+  'libSBRenc/src/env_est.h',
+  'libSBRenc/src/fram_gen.h',
+  'libSBRenc/src/invf_est.h',
+  'libSBRenc/src/mh_det.h',
+  'libSBRenc/src/nf_est.h',
+  'libSBRenc/src/ps_bitenc.h',
+  'libSBRenc/src/ps_const.h',
+  'libSBRenc/src/ps_encode.h',
+  'libSBRenc/src/ps_main.h',
+  'libSBRenc/src/resampler.h',
+  'libSBRenc/src/sbr.h',
+  'libSBRenc/src/sbr_def.h',
+  'libSBRenc/src/sbr_misc.h',
+  'libSBRenc/src/sbrenc_freq_sca.h',
+  'libSBRenc/src/sbrenc_ram.h',
+  'libSBRenc/src/sbrenc_rom.h',
+  'libSBRenc/src/ton_corr.h',
+  'libSBRenc/src/tran_det.h',
+  'libPCMutils/src/version.h'
+)
+
+fdk_aac_headers = [
+  'libAACdec/include/aacdecoder_lib.h',
+  'libAACenc/include/aacenc_lib.h',
+  'libSYS/include/FDK_audio.h',
+  'libSYS/include/genericStds.h',
+  'libSYS/include/machine_type.h',
+  'libSYS/include/syslib_channelMapDescr.h',
+]
+
+install_headers(files(fdk_aac_headers), subdir : 'fdk-aac')
+
+# Make using fdk-aac as a subproject behave the same as when used installed.
+subdir('fdk-aac')
+
+cc = meson.get_compiler('c')
+m_dep = cc.find_library('m', required : false)
+
+extra_args = ['-Dfdk_aac_EXPORTS']
+if get_option('remove-date').enabled()
```

Comment:
What is the point of using .enabled() instead of a boolean, when auto and disabled are identical?

Reply:
There is one difference, you can control the value with -Dauto_feautures. Dunno if that was the intention of @thiblahute. As he's a gstreamer dev, I think he just used feature because we use that everywhere in gst even for simple booleans.

Reply:
`--auto-features=disabled` is very handy...

Reply:
This is not a feature. It makes no sense to disable it.

Opinionated statement: Gst is wrong for using this pattern. Please submit a ticket to meson for adding --default-bools-to-false.

Objective statement that cannot be argued with: this neither enables nor disables any part of the build, so even misusing --auto-features serves no purpose here. This controls whether to replace the `__DATE__` and `__TIME__` macros with an empty string rather than the default compiler behavior which is to   ignore both date and time and use the value of $SOURCE_DATE_EPOCH instead.

Your CI is not benefiting from this.

Reply:
Also it seems that cmake and autotools both don't offer this option, but Android.bp always defines it unconditionally... huh.

Reply:
Tbh I would just drop that weird option for now. Indeed it does not seems to fit a feature, unlike build_programs that do make sense as a feature.

Reply:
Agreed, it is not a feature, I overthought this one. It allows making my build reproducible so I still need it, will make it a bool

Reply:
Well, it is building on debian, and iiuc the code requires to support it, which is not the case here.

I guess it is good enough to add an option no?

Reply:
It's fine to add an option.

Anyway, some compilers automatically support SOURCE_DATE_EPOCH (gcc does), but not all. So it might or might not be a general solution for you, but if it is, you don't need the code to support it.

You just need the code to use `__DATE__` rather than idk, the build system adding the compiler flag `-DBUILD_DATE="$(/usr/bin/date)"`.

Yes, there are people doing the latter. (!!!)